styleguide/angular.md

```diff
@@ -0,0 +1,178 @@
+
+Components provide a way cleaner interface than directives did in earlier versions of Angular!
+
+> **Rule of thumb:** Don't write directives! Refactor your directives towards a component. As soon as one has done it once, it will be straight forward (..._most of the time_).   
+
+> **To Clarify:** ~~Where do we manipulate the DOM in components?~~ DOM manipulations happen during the `$postLink` lifecycle hook.  
+
+1. Forget about scope inheritance! It is evil. All components have an isolated scope! All the data that is passed into the component is clearly defined in the `bindings` section of the components definition.
+2. Create services to share logic and load data if necessary.
+
+### Structure
+Reworking a codebase towards components brings up an old discussion about how to structure Angular code. Some prefer to group the code together based on their type of content (_e.g._ having a `service` folder, a `directive` folder ...). 
+
+Even though this might be subjective, the most :
+
+	- `app.js`
+	- `/user`
+		- `userProfile.component.js`
+		- `userProfile.template.html`
+		- `userProfileForm.component.js`
+		- `userProfileForm.template.html`
+		- `userService.service.js`
+		- `userStatusFilter.filter.js`
+- `/common`
+
+How and where your application is served from depends on the project and the rest of the tech stack.
+
+> **Discuss:** We are currently only working using one single module per application. We should probably think about that! It prevents us from sharing code between the website and the back office application.  
+
+### Controllers  and Timing
+
+When using components one can _safely_ write the controllers as ES6 classes (depending on the projects requirements in terms of browser support [Can I use... Support tables for HTML5, CSS3, etc](https://caniuse.com/#feat=es6-class)).
+
+> **Note:** Safely means that I did not encounter cases where it did not work (in contrast to plain directives). Further, we do cross-compile the source code via Babel for IE11 support.  
+
+Components (_i.e._ their controllers) have a well defined [interface for lifecycle hooks](https://code.angularjs.org/1.6.10/docs/guide/component#component-based-application-architecture).  To get  the timing right one can follow a simple template:
+
+```Javascript
+class MyComponentController {
+	  /**
+     * Dependencies defined in $inject are passed here.
+     */
+    constructor($timeout, someService) {
+        // assign the dependencies to your controller
+        this.$timeout = $timeout;
+        this.someService = someService;
+    }
+    /**
+     * $onInit is called if the bindings are resolved.
+     * Even though Angular will not `await` your $onInit method you can use async here to simplify your code and control the timing using a guard.
+	   */
+    async $onInit() {
```

Comment:
since angular doesn't await what will happen with errors?

Reply:
Good point! I forgot the error handling.